### PR TITLE
PLT-6905 - Updating channel header design

### PR DESCRIPTION
**Pull request by asaadmahmood**
#### Summary
PLT-6905 - Updating channel header design

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-6905

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Added or updated unit tests (required for all new features)
- [ ] Added API documentation (required for all new APIs)
- [ ] All new/modified APIs include changes to the drivers
- [ ] Has enterprise changes (please link)
- [x] Has UI changes
- [ ] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates
- [ ] Touches critical sections of the codebase (auth, upgrade, etc.)

### CODE REVIEW

**Comment by jasonblais** [comment]
@asaadmahmood Tested it out. Sorry if some of this is duplicate from pre-release.

Bugs:
1) No active state for pin icon
image

2) Recent mentions icon doesn't work (it only works if you already have RHS open)
3) "Add a channel description" below channel name doesn't work
4) Channel header truncation is better, but there's still some at certain screen sizes
image

Feedback/thoughts:

5) I feel the count in channel member icon could be in grey. I'm not sure it needs to be blue.
image

6) Do we want the "Add a channel description" to be a link colour? I worry if it's a distraction in the long term (in channels that don't need/have a header)
image

7) Is the "v" centered with the channel name?
image

8) I feel the icons are really big compared to the sidebar icons. Feel it's really on your face. (NOTE: It could also be that I'm not used to the new icons, so my eyes are drawn to them because of that)
image

9) Is the flag icon centered inside the circle?
image

10) Should the search bar expand once you put focus on it, rather than after you hit Enter to get the search results?

**Comment by jasonblais** [comment]
11. Whitespace between search bar and icons after you expand RHS
image

12) Is the search icon centered inside the circle in mobile view?
image

13) Should the "Members" title/heading have a faint grey background in light themes to differentiate from the user list (and something else in dark themes)
image

14) Not sure if this is expected? (in shrinked desktop view)

image

**Comment by asaadmahmood** [comment]
1. Yup that's from master.
2. Probably from master as well, nothing specific to this PR I think.
3. For Saturnino to fix.
4. Will fix.
5. We can come up with a decision for that, though I agree, it if it's gray, it might be hard to read, so maybe just the center channel color.
6. 0/5.
7. Might need to move down a a bit.
8. I think they look fine, they're also pretty similar size to the icons we have in our current header.
9. Pretty much horizontally, but it may need to move to the top to center vertically, but since it's heavy from the top, I don't suggest that.
10. Maybe.
11. Will fix.
12. Pretty much, but can be moved to the right by 1 px, as the icon is heavy from the left.
13. Can be, but wasn't in the design.
14. Yeah I think we can probably stop truncating if the sidebar is overlapping.

**Comment by lfbrock** [comment]
screenshot 2017-06-29 22 34 09

Another channel header truncation case when I open the RHS: 
screenshot 2017-06-29 23 05 01

**Comment by lfbrock** [comment]
Group message header broken: 
screenshot 2017-06-30 00 43 25

Mobile/tablet view menus have a gap: 
screenshot 2017-06-30 00 43 53

**Comment by asaadmahmood** [comment]
@lfbrock The border bottom for the header is 2 pixels in the design. We can make it one pixel though. But I personally am okay with it being 1 pixel as well, as it provides better separation.

**Comment by lfbrock** [comment]
@asaadmahmood - okay yeah, I think we should make it 1 pixel then to be consistent with our other lines

**Comment by jwilander** [comment]
This has a bad rebase, there's a ton of files changed here that have nothing to do with the channel header. Let me know if you need help fixing

**Comment by Hanzei** [comment]
The position of the member icon is broken

screenshot from 2017-07-05 17-34-05

**Comment by jasonblais** [comment]
Thanks @Hanzei, great find! Let us know if you find anything else

**Comment by Hanzei** [comment]
On the Spinmint server clicking on "Recent Mentions" does nothing. Is this intended?

File: webapp/components/channel_header.jsx [hunk 1]
```diff
@@ -759,10 +772,51 @@ export default class ChannelHeader extends React.Component {
         }
 
         let headerText;
-        if (this.state.enableFormatting) {
-            headerText = TextFormatting.formatText(channel.header, {singleline: true, mentionHighlight: false, siteURL: getSiteURL()});
-        } else {
+        let headerTextContainer;
+        if (channel.header) {
             headerText = channel.header;
```
Comment (ccbrown) [inline webapp/components/channel_header.jsx]: Is raw HTML intentionally allowed in channel headers when formatting is disabled?

Comment (ccbrown) [inline webapp/components/channel_header.jsx]: Disregard my last comment. We'll patch it in another PR.

**Comment by jasonblais** [comment]
@Hanzei Not intended but it should be fixed on the `master` branch.

**Comment by jwilander** [comment]
I need to push one small change to this and then I'll merge it

**Comment by Hanzei** [comment]
@jasonblais are you sure? On pre-release it doesn't work neither

**Comment by jasonblais** [comment]
Yeah, we filed a ticket for it..thanks @Hanzei for following up on it 👍

**Pull request merged**